fail2ban/server/datedetector.py

```diff
@@ -165,8 +165,8 @@ def _cacheTemplate(self, template):
 		r"%b %d, %ExY %I:%M:%S %p",
 		# ASSP: Apr-27-13 02:33:06
 		r"^%b-%d-%Exy %k:%M:%S",
-		# 20050123T215959, 20050123 215959, 20050123  85959
-		r"%ExY%Exm%Exd(?:T|  ?)%ExH%ExM%ExS(?:[.,]%f)?(?:\s*%z)?",
+		# 20050123T215959, 20050123 215959, 20050123  85959, 20050123-21:59:59
+		r"%ExY%Exm%Exd(?:-|T|  ?)%ExH:?%ExM:?%ExS(?:[.,]%f)?(?:\s*%z)?",
```

Comment:
Not against this enhancement, but...
Would not an extra datepattern in filter help too? I mean something like that instead (or together with that):
```ini
datepattern = ^\[?%%ExY%%Exm%%Exd[-|T]%%ExH:?%%ExM:?%%ExS(?:[.,]%%f)?(?:\s*%%z)?\]?
              ^{DATE}
```
This would also allow to use that filter with older fail2ban versions.
I'll try to adjust the filter a bit.

Reply:
Filter extended a bit now.

Reply:
Does the `datepattern` that you've added to the filter extend (rather than replace) what's available from `datedetector.py`? Particularly, I'm asking about with the `^{DATE}` part included. If so, I like this better than messing with `datedetector.py` but I'd want to change the expression to be more particular to xrdp's format.

Reply:
`^{DATE}` is the same as default pattern but anchored at begin of line only (default datepatterns would prefer then but use both variants, anchored at begin of line and word boundaries).
As you can see the tests passed and 2nd date format in the tests also found.